tsconfig.json

```diff
@@ -1,9 +1,10 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es6",
+    "useDefineForClassFields": true,
     "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
-    "skipLibCheck": false,
+    "allowJs": false,
+    "skipLibCheck": true,
```

Comment:
Why change this? We should not skip the lib check.

Reply:
In fact, we should. There are a lot of type errors in underlying libraries that we cannot solve.

Reply:
The problem with enabling `skipLibCheck` is that we instruct the TypeScript compiler to skip type checking for *all* declaration (.d.ts) files, including our own `vite-env.d.ts`, which can lead to runtime errors.

> There are a lot of type errors in underlying libraries that we cannot solve.

We have always had this enabled in the wallet and in the bridge and never faced the issue you mention with a third-party lib.

Reply:
The problem with enabling `skipLibCheck` is that we instruct the TypeScript compiler to skip type checking for *all* declaration (.d.ts) files, including our own `vite-env.d.ts`, which can lead to runtime errors.

> There are a lot of type errors in underlying libraries that we cannot solve.

We have always had this enabled in the wallet and in the bridge and never faced the issue you mention with a third-party lib.

Reply:
No, [we can't](https://www.typescriptlang.org/tsconfig#exclude).

I see that this is indeed an issue with vite. Reverting `skipLibCheck` to `false` a lot of errors appear.

Also, including `src/vite-env.d.ts` does not seem to work 🤔

Reply:
pfff... https://github.com/microsoft/TypeScript/issues/30511